tests/languages/json-source-code.test.js

```diff
@@ -497,4 +497,302 @@ describe("JSONSourceCode", () => {
 			});
 		});
 	});
+
+	describe("getTokenOrCommentBefore()", () => {
+		let sourceCode;
+		let file;
+		let language;
+		let parseResult;
+
+		beforeEach(() => {
+			file = { body: '{ "a": 1 }', path: "test.json" };
+			language = new JSONLanguage({ mode: "json" });
+			parseResult = language.parse(file);
+			sourceCode = new JSONSourceCode({
+				text: file.body,
+				ast: parseResult.ast,
+			});
+		});
+
+		it("should return null for the first token", () => {
+			const firstToken = sourceCode.ast.tokens[0];
+			assert.strictEqual(
+				sourceCode.getTokenOrCommentBefore(firstToken),
+				null,
+			);
+		});
+
+		it("should return the previous token for a middle token", () => {
+			const tokens = sourceCode.ast.tokens;
+			const secondToken = tokens[1];
+			assert.strictEqual(
```

Comment:
```suggestion
			assert.deepStrictEqual(
```

Could you take a look at all the `strictEqual` assertions?

Since `Token` types always return objects like the example below, I believe it would make more sense to use `deepStrictEqual` for all the test cases.

```js
{
  type: 'LBrace',
  loc: {
    start: { line: 1, column: 1, offset: 0 },
    end: { line: 1, column: 2, offset: 1 }
  },
  range: [ 0, 1 ]
}
```